src/client/src/components/EditUserProfile.jsx

```diff
@@ -20,35 +21,68 @@ const EditUserProfile = () => {
     education: [],
     skills: [],
     workExp: [],
-    volunteering: []
+    volunteering: [],
   });
 
+  //Global loginState
+  const [login, setLogin] = useContext(Context);
+
+  useEffect(() => {
+    fetchSession();
+  }, []);
+
+  //Having the loginState persist on all pages
+  const fetchSession = async () => {
+    try {
+      setLogin({
+        isLoggedIn: true,
+      });
+    } catch (error) {
+      console.log(error);
+    }
+  };
```

Comment:
If someone tries to access any page without having an account, the server should catch that error and log it onto the console. How would a person perceive this in their web browser?

Reply:
i added the logic for usertime, userprofile and edituserprofile, not if a user is not logged in, the website will output "user needs to login"